src/main/scala/sigmastate/serialization/ValueSerializer.scala

```diff
@@ -17,45 +17,83 @@ trait ValueSerializer[V <: Value[SType]] extends SigmaSerializer[Value[SType], V
   }
 }
 
-object ValueSerializer extends SigmaSerializerCompanion[Value[SType]] {
+trait OpCodes {
+
   type OpCode = Byte
-  type Tag = OpCode
 
-  val TaggedVariableCode = 1: Byte
+  val TaggedVariableCode:   OpCode = 1: Byte
+
+  // EvaluatedValue descendants
+
+  val IntConstantCode:          OpCode = 11: Byte
+  val TrueCode:                 OpCode = 12: Byte
+  val FalseCode:                OpCode = 13: Byte
+  val UnitConstantCode:         OpCode = 14: Byte
+  val BigIntConstantCode:       OpCode = 15: Byte
+  val ByteArrayConstantCode:    OpCode = 16: Byte
+  val GroupElementConstantCode: OpCode = 17: Byte
+
+  // Relation descendants
+
+  val LtCode:     OpCode = 21: Byte
+  val LeCode:     OpCode = 22: Byte
+  val GtCode:     OpCode = 23: Byte
+  val GeCode:     OpCode = 24: Byte
+  val EqCode:     OpCode = 25: Byte
+  val NeqCode:    OpCode = 26: Byte
+
+  val ConcreteCollectionCode: OpCode = 35: Byte
 
-  val IntConstantCode = 11: Byte
-  val TrueCode = 12: Byte
-  val FalseCode = 13: Byte
-  val UnitConstantCode = 14: Byte
-  val ByteArrayConstantCode = 15: Byte
+  val TupleCode:  OpCode = 36: Byte
+  val AndCode:    OpCode = 37: Byte
+  val OrCode:     OpCode = 38: Byte
+  val NotCode:    OpCode = 39: Byte
 
-  val LtCode = 21: Byte
-  val LeCode = 22: Byte
-  val GtCode = 23: Byte
-  val GeCode = 24: Byte
-  val EqCode = 25: Byte
-  val NeqCode = 26: Byte
+  // TwoArgumentsOperation descendants
 
-  val ConcreteCollectionCode = 35: Byte
-  val TupleCode = 36: Byte
-  val AndCode = 37: Byte
-  val OrCode = 38: Byte
+  val MinusCode:          OpCode = 40: Byte
+  val PlusCode:           OpCode = 41: Byte
+  val XorCode:            OpCode = 42: Byte
+  val AppendBytesCode:    OpCode = 43: Byte
+  val ExponentiateCode:   OpCode = 44: Byte
+  val MultiplyGroupCode:  OpCode = 45: Byte
+}
+
+object ValueSerializer
+  extends SigmaSerializerCompanion[Value[SType]]
+  with OpCodes {
+
+  override type OpCode = Byte
```

Comment:
Overriding `OpCode`? Why? I'm ask cause it has the same type as the inherited one - `Byte`. I suggest to remove `OpCode` from Value seialaizer, make OpCodes as an object (and move it to it's own file). Then you can eaily get this type as `OpCodes.OpCode` or `import OpCodes._`

Reply:
I like the idea. Done.